cabot/cabotapp/tasks.py

```diff
@@ -139,3 +154,50 @@ def clean_db(days_to_retain=60):
 
     clean_db.apply_async(kwargs={'days_to_retain': days_to_retain},
                          countdown=3)
+
+
+@task(ignore_result=True)
+def send_schedule_problems_email(schedule_id):
+    """
+    Send off an email as a celery task
+    :param schedule_id schedule ID
+    """
+    try:
+        schedule = models.Schedule.objects.get(pk=schedule_id)
+        problems = schedule.problems
+    except models.Schedule.DoesNotExist, models.ScheduleProblems.DoesNotExist:
+        # if the schedule or problems got deleted, nothing to do
+        return
+
+    # check if problems became silenced since the email got queued
+    if problems.is_silenced():
+        return
+
+    # build the message
+    # make the schedule link absolute (add domain name) because this is going into an email
+    cabot_schedule_url = build_absolute_url(schedule.get_edit_url())
+
+    # build links to snooze email alerts
+    snooze_hours = [4, 12, 24]
```

Comment:
put this in [defs](https://github.com/Affirm/cabot/blob/master/cabot/cabotapp/defs.py)? since it's a constant

Reply:
Done :+1: